sgv2-service-common/src/main/java/io/stargate/sgv2/common/grpc/DefaultStargateBridgeClient.java

```diff
@@ -90,15 +96,16 @@ public Response executeBatch(Batch request) {
   }
 
   @Override
-  public CqlKeyspaceDescribe getKeyspace(String keyspaceName) throws UnauthorizedKeyspaceException {
-    if (!authorizeSchemaRead(SchemaReads.keyspace(keyspaceName, SchemaRead.SourceApi.REST))) {
+  public Optional<CqlKeyspaceDescribe> getKeyspace(String keyspaceName)
+      throws UnauthorizedKeyspaceException {
+    if (!authorizeSchemaRead(SchemaReads.keyspace(keyspaceName, sourceApi))) {
       throw new UnauthorizedKeyspaceException(keyspaceName);
     }
     return getAuthorizedKeyspace(keyspaceName);
   }
 
-  private CqlKeyspaceDescribe getAuthorizedKeyspace(String keyspaceName) {
-    return stripTenantPrefix(schema.getKeyspace(addTenantPrefix(keyspaceName)));
+  private Optional<CqlKeyspaceDescribe> getAuthorizedKeyspace(String keyspaceName) {
+    return Optional.of(stripTenantPrefix(schema.getKeyspace(addTenantPrefix(keyspaceName))));
```

Comment:
I am thinking that the whole Tenant prefix stuff, should be extracted to a "util" pojo class that can accept the tenant id and then provide methods like `stripTenantPrefix` and `addTenantPrefix`.. It can be tested good in the unit test and reused where needed..

These lines (https://github.com/stargate/stargate/pull/1634/files#diff-da5eacd2c92cb5e4fa71fa2de0240eed08a2dcd1ad8287cdebc4abe34edc04c9R199-R223) seems as the candidate for extraction here..